pkg/generator/pool.go

```diff
@@ -90,13 +90,21 @@ func NewParameterPool[T any](
 
 // Get returns a new parameter. It is fetched from the pool or generated if the
 // pool is empty.
-func (pp *ParameterPool[T]) Get() (*T, error) {
-	generated := <-pp.pool
-	err := pp.persistence.Delete(generated)
-	if err != nil {
-		return nil, fmt.Errorf("could not delete persisted parameter: [%w]", err)
+func (pp *ParameterPool[T]) Get(ctx context.Context) (*T, error) {
+	select {
+	case generated := <-pp.pool:
+		err := pp.persistence.Delete(generated)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"could not delete persisted parameter: [%w]",
+				err,
+			)
+		}
+
+		return generated, nil
+	case <-ctx.Done():
+		return nil, fmt.Errorf("context is done")
```

Comment:
Wouldn't it be clearer to do something like:
```
func (pp *ParameterPool[T]) GetNow() (*T, error) {
	select {
	case generated := <-pp.pool:
		err := pp.persistence.Delete(generated)
		if err != nil {
			return nil, fmt.Errorf(
				"could not delete persisted parameter: [%w]",
				err,
			)
		}

		return generated, nil
	default:
		return nil, fmt.Errorf("pool is empty")
	}
}
```

In `Execute` function, we do not want to wait for the pool even one minute. The scheduler stopped the generation process so there is no chance something will appear in the channel. I think this will also work better once we move getting the parameters to phase 2 of TSS: we will check `pool.CurrentSize()` before `Execute()` call and later, we'll use `GetNow()`.

Reply:
Makes sense, see https://github.com/keep-network/keep-core/pull/3212/commits/ea027cfd85072742f350792d5dfbc1d998c74cac